### ISSUE TITLE
Question on Spanish dataset compilation

### ISSUE THREAD

**Issue by simon-clematide** [coastalcph/histnorm#2]
A small question regarding the compilation of the Spanish training dataset. Were the 4 parts just concatenated on their historical order (this probably matters for the training size ablation experiments, e.g. if your training material is mostly 16th's century, but most of your test/dev set material is from 17th-19th century. Or were  the size-reduced samples not continuous chunks?

**Comment by mbollmann** [comment]
Cf. [Bollmann (2018)](https://www.linguistics.rub.de/forschung/arbeitsberichte/22.pdf), p.37:

> From each century, 80% of all sentences are randomly selected for the training set, another 10% for the development set, and the final 10% for the test set of their respective language. This procedure is supposed to somewhat balance the large time period covered by the corpus.

No attempt was made to further investigate the effect of using data from different centuries, mainly because my focus was on a comparison across languages, and most datasets didn't have this information.

**Comment by simon-clematide** [comment]
Thanks for the answer and the reference.  Am I correct that for the size ablation experiments, let's say the ones with 100 tokens, the training material was then only from the 16th century?

**Comment by simon-clematide** [comment]
I just reread the Bollmann 2019 footnote: "12The ten training splits consist of chunks of n tokens
that are spaced equidistantly across the full training set; for
larger n, the number of chunks is reduced so that no splits
overlap to more than 50%." Then the only remaining question is whether the training material was randomized or not.

**Comment by mbollmann** [comment]
They were not shuffled before producing the splits, which should be equivalent to using `scripts/split_sample.py --continuous` from this repo.

That means for the 100-tokens scenario, it is likely that each training split contained samples from a particular century only, but each century was represented in *some* of the ten splits.

**Comment by simon-clematide** [comment]
Ok, do you per chance still have the exact low-resource splits as files? I imagine that the random selection of the splits for the 100 token case, for instance, has some impact on the results.

**Comment by mbollmann** [comment]
There is no random selection. `scripts/split_sample.py --continuous -n 10 -s 100` produces the exact splits I used, reproducibly. (I do still have the exact files, which is how I verified this.)

**Comment by simon-clematide** [comment]
Ok, thanks for your answer and sorry for not thinking enough by myself.  I think I understand now, `--continuous` keeps everything in order and the number of subsamples defines the intervals from where the data is drawn.
In ASCII art this looks like that:
```
16th...|17th.......|18th.........................|19th......................................................  # concatenation of the data
|          |          |          |          |          |          |          |          |          |                            # Equal splits of the data
.          .          .          .          .          .          .          .          .          .                              # Samples of size 100                 
```
 Even if for instance Spanish 16th is small and 19th is a lot larger, there would be only 1 out of 10 subsamples of Spanish 16th. Given that only 1 model out of 10 is then used for the dev/test sets, which again are dominated by the 18th/19th century, its poor performance will not influence the averaged performance that much.

**Comment by mbollmann** [comment]
That looks like a nice visualization I could have included in the Appendix :)

But yes, this looks and sounds correct to me.
